### PR TITLE
Add font-weight to docs for button

### LINKED ISSUE #4213: Consider adding font and opacity as Button Kind theming options
### Expected Behavior

`font` and `opacity` Button styling documentation and/or reference to  in `base.js`

### Actual Behavior

Button font-weight changed unexpectedly as reported in https://github.com/grommet/grommet-theme-hpe/issues/80

Resolution was to add `button.primary.font.weight` and `button.secondary.font.weight` to theme, however when investigating, I was unable to see that `font` (as well as `opacity`) was a property in documentation and/or `base.js` but is referenced when implementing styling: https://github.com/grommet/grommet/blob/2684f256aefe2b15b659459cf542b36489fa7257/src/js/components/Button/StyledButtonKind.js#L121

### URL, screen shot, or Codepen exhibiting the issue

https://github.com/grommet/grommet-theme-hpe/issues/80

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: Stable
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR adds font weight to the docs for `button` to show users that they can
Customize the font weight with the different types of `buttons`
#### Where should the reviewer start?
docs.js
#### What testing has been done on this PR?
storybook 
#### How should this be manually tested?
you can add something like 
```
button: {
  primary: {
   font: { weight: '900'}
  }
}

```

as well as `secondary` and `default`
#### Any background context you want to provide?
https://github.com/grommet/grommet-theme-hpe/pull/81
https://github.com/grommet/grommet/issues/4213

There was also a note to add opacity however this is already listed under 
https://v2.grommet.io/button#button.default.background.opacity
for all `primary` `secondary` and `default`
#### What are the relevant issues?
closes #4213 
#### Do the grommet docs need to be updated?
auto
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Button/doc.js [hunk 1]
```diff
@@ -274,6 +274,10 @@ export const themeDoc = {
     description: `The color of the label for default buttons.`,
     type: 'string | { dark: string, light: string }',
   },
+  'button.default.font.weight': {
+    description: `The weight of the text label for default buttons.`,
+    type: 'string',
```
Comment (ShimiSun) [inline src/js/components/Button/doc.js:279]: I think this should accept a number as well, can you verify?

File: src/js/themes/base.js [hunk 2]
```diff
@@ -415,13 +415,19 @@ export const generate = (baseSpacing = 24, scale = 6) => {
...
       //   padding: {
       //     vertical: undefined,
       //     horizontal: undefined,
       //   },
       //   extend: undefined,
       // },
       // primary: {
+      //   font: {
+      //     weight: undefined,
+      //   },
```
Comment (ShimiSun) [inline src/js/themes/base.js:430]: Good structure.

File: src/js/components/Button/doc.js (hunk 1 above)
Comment (britt6612) [reply src/js/components/Button/doc.js:279]: yes I will verify I just added a number in a string but I will do a number

**Comment by britt6612** [comment]
> Is this functionality already works as expected and the only piece that was missing was the documentation?

Yes this already works as expected.. 
for example were using it in design-system
https://github.com/grommet/grommet-theme-hpe/pull/81

File: src/js/components/Button/doc.js (hunk 1 above)
Comment (britt6612) [reply src/js/components/Button/doc.js:279]: you are right I just checked with 
```
      font: { weight: 800 }
```

File: src/js/components/Button/doc.js [hunk 3]
```diff
@@ -394,6 +398,10 @@ export const themeDoc = {
     description: `The color of the label for primary buttons.`,
     type: 'string | { dark: string, light: string }',
   },
+  'button.primary.font.weight': {
+    description: `The wight of the text label for default buttons.`,
```
Comment (ShimiSun) [inline src/js/components/Button/doc.js:402]:
```suggestion
    description: `The wight of the text label for primary buttons.`,
```

Comment (halocline) [reply src/js/components/Button/doc.js:402]: Typo "weight" instead of "wight"

Comment (britt6612) [reply src/js/components/Button/doc.js]: @halocline good catch!

Comment (ShimiSun) [reply src/js/components/Button/doc.js]: Meet https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker , it is a good friend of mine :)

**Pull request merged**